samples/helm-deploy/README.md

```diff
@@ -0,0 +1,162 @@
+# How-To: Deploy to Kubernetes using Helm
+
+## Setup Dev Environment (optional)
+
+The easiest way to setup your environment is to install [Kind](https://kind.sigs.k8s.io/) or [Minikube](https://minikube.sigs.k8s.io/docs/start/). It is also valid to setup a cloud provider kubernetes cluster (i.e. AKS, EKS, GKE, ...)
+
+## Push an image to the Docker registry
+
+The following steps illustrates how to push a webservice image in order to be
+deployed in Kubernetes clsuter.
+
+1. Build an image using the following
+   [Dockerfile](.../../../../src/CarbonAware.WebApi/src/Dockerfile)
+
+   ```sh
+   cd <Dockerfile path>
+   docker build -t <registry>/myapp:v1 .
+   ```
+
+1. Login into docker using the username and password credentials that are needed in
+   order to push.
+
+   ```sh
+   docker login 
+   ```
+
+1. Push to Docker registry 
+
+   ```sh
+   docker push <registry>/myapp:v1
+   ```
+
+
+## Create a new Helm chart (optional)
+
+Run `helm` to ensure you have the helm CLI running properly. Then you can create
+a new helm chart by running
+
+```bash
+helm create <chart-name>
+```
+
+## Setting up the Helm chart
+
+Once you've got your helm chart open (whether from scratch or existing), the
+main files you will likely be working with are `Chart.yaml` and `values.yaml`.
+
+### Chart.yaml
+
+In `Chart.yaml`, we won't need to make any changes but make note of the chart
+`name`, as you will need to reference it in commands later on.
+
+### Values.yaml
+
+In `values.yaml`, we need to change a couple fields. In the `image` section, you
+will need to set
+
+- The `repository` field to be `<image-repository>`
+- The `pullPolicy` field to be `IfNotPresent` (pulls a new image if not present)
+  or `Always` (pulls a new image every time).
+- The `tag` field if you need a particuar tag version
+
+Set the `nameOverride` and `fullNameOverride` fields to make it easier to
+reference your helm chart, and ensure they are not the same.
+
+In the `serviceAccount` section, ensure that
+
+- The `name` field is set to the name of the helm chart from `Chart.yaml`.
+
+In the `monitorConfig` section, you will need to adjust the `liveness` and
+`readiness` that the helm chart will ping to ensure the sdk is ready. As a
+default, you should set the path to `/health`.
```

Comment:
Where is `monitorConfig`? Liveness / Startup probe should be set into the deployment yaml.